### DISCUSSION TITLE
Help needed with development environment setup v0.11.0

### DISCUSSION THREAD

**Discussion by munkizh** [cncf/landscape2#749] [Q&A]
Hi everyone! I'm trying to set up the dev environment for this project but running into some configuration issues.

### What I've tried:
1. Installed Node and Yarn
2. Ran `yarn install` which installed the dependencies in package.json
3. Tried `cargo build` for the Rust components
4. Tried `npm install` just in case yarn wasn't the way to go

### Current issues:
1. TypeScript configuration seems to be breaking - the tsconfig.node.json file has a moduleResolution set to "bundler" which isn't compatible with the current TypeScript version
2. Not sure about the correct sequence for setting up both the JavaScript and Rust parts of the project
3. I am also getting import errors from pathing. Ex: `[plugin:vite:import-analysis] Failed to resolve import "../../wasm/overlay/landscape2_overlay" from "src/utils/overlayData.ts"  Does the file exist?`

### Environment details:
- Node & Yarn installed
- TypeScript (5.6.3)

### Questions:
1. What's the correct order of setup steps?
2. Are there specific version requirements for Node? 
3. Do I need to build the Rust components before running the UI parts?

Any guidance would be greatly appreciated! Let me know if you need any additional information.

**Comment by tegioz** [comment]
Hi @munkizh 👋 

To build the landscapes generator from the source, you should only need to run `cargo build`. The CLI tool crate includes a [build.rs script](https://github.com/cncf/landscape2/blob/4d6e63e505cb8bd9a2a359b54fc5a1e62c581bb0/crates/cli/build.rs) that will build all UI components when needed. So you shouldn't need to build them in advance or separately. Please note that you'll need to have `yarn` and `wasm-pack` installed, as the build script will need them.
